bin/si-sdf-api-test/tests/5-emulate_paul_stack.ts

```diff
@@ -26,155 +20,110 @@ async function emulate_paul_stack_inner(
+  let schemaVariants = await getVariants(sdf, changeSetId);
+  let createRegionBody = createComponentPayload(schemaVariants, "Region");
 
-  const awsRegionVariant = await extractSchemaVariant(
-    schemaVariants,
-    "Region",
-    "AWS",
-  );
-  const awsRegionVariantId = awsRegionVariant.schemaVariantId;
+  // Get the views and find the default one
+  const views = await getViews(sdf, changeSetId);
+  const defaultView = views.find((v: any) => v.isDefault);
+  assert(defaultView, "Expected to find a default view");
 
-  const _diagram = await getDiagram(sdf, changeSetId);
-  //await sleepBetween(2000, 10000);
-
-  // CREATE COMPONENTS
-  // create region component
+  // Create a region component
   const regionComponentId = await createComponent(
     sdf,
     changeSetId,
-    awsRegionVariantId,
-    0,
-    0,
-    undefined,
-    newCreateComponentApi,
-  );
-
-  //await sleepBetween(3000, 6000);
-
-  await setComponentGeometry(
-    sdf,
-    changeSetId,
-    regionComponentId,
-    0,
-    0,
-    1800,
-    800,
-  );
-  //await sleepBetween(1000, 5000);
-
-  // UPDATE REGION
-  const regionValue = "us-east-1";
-  const { values: RegionPropValues } = await getPropertyEditor(
-    sdf,
-    changeSetId,
-    regionComponentId,
+    defaultView.id,
+    createRegionBody,
   );
-  // await sleep(2000);
-
-  {
-    const { attributeValueId, parentAttributeValueId, propId } =
-      attributeValueIdForPropPath(
-        "/root/domain/region",
-        awsRegionVariant.props,
-        RegionPropValues,
-      );
-
-    await setAttributeValue(
-      sdf,
+  // update region prop
+  await sdf.call({
+    route: "attributes",
+    routeVars: {
+      workspaceId: sdf.workspaceId,
       changeSetId,
-      regionComponentId,
-      attributeValueId,
-      parentAttributeValueId,
-      propId,
-      regionValue,
-    );
-  }
+      componentId: regionComponentId,
+    },
+    body: {
+      "/domain/region": "us-east-1",
+    },
+  });
 
-  // await sleepBetween(5000, 15000);
 
-  // CREATE VPC
-  const vpcVariant = await extractSchemaVariant(
-    schemaVariants,
-    "VPC",
-    "AWS EC2",
-  );
-  const vpcVariantId = vpcVariant.schemaVariantId;
 
+  // CREATE VPC
+  let createVPCBody = createComponentPayload(schemaVariants, "AWS::EC2::VPC");
   const vpcComponentId = await createComponent(
     sdf,
     changeSetId,
-    vpcVariantId,
-    0,
-    0,
-    undefined,
-    newCreateComponentApi,
+    defaultView.id,
+    createVPCBody,
   );
 
-  // await sleepBetween(1000, 2000);
+  // subscribe vpc to region component
+  const subResponse = await sdf.call({
+    route: "attributes",
+    routeVars: {
+      workspaceId: sdf.workspaceId,
+      changeSetId,
+      componentId: vpcComponentId,
+    },
+    body: {
+      "/domain/extra/Region": {
+        "$source": {
+          "component": regionComponentId,
+          "path": "/domain/region",
+        }
+      },
+    },
+  });
```

Comment:
It's nice to see it written out like this since I usually see it in pieces across Vue components and composables. This looks good.